app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -97,18 +99,38 @@ class BrowserActivity : DuckDuckGoActivity(), CoroutineScope by MainScope() {
             renderer.renderBrowserViewState(it)
         })
         viewModel.awaitClearDataFinishedNotification()
+
+        keyboardVisibilityNotifier = KeyBoardVisibilityNotifier(this)
+
+    }
+
+
+    override fun onResume() {
+        keyboardVisibilityNotifier.register(rootViewGroup,::onKeyboardStateChanged)
+        super.onResume()
     }
 
+    private fun onKeyboardStateChanged(isOpen: Boolean) {
+        currentTab?.keyboardStateChanged(isOpen)
+    }
+
+    override fun onPause() {
+        keyboardVisibilityNotifier.unRegister(rootViewGroup)
+        super.onPause()
+    }
     override fun onStop() {
+        keyboardVisibilityNotifier.unRegister(rootViewGroup)
```

Comment:
I don't think we should put the same `unRegister` method call in all these 3 lifecycle methods. Usually, you want to unregister from the opposite lifecycle method where you registered the listener. That will mean that if you register `onResume` you unregister at `onPause`.